src/rt-app.c

```diff
@@ -913,174 +921,203 @@ static int __sched_priority(sched_data_t *sched_data)
 	 return 0;
 }
 
-static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
+
+static void __log_policy_priority_change(thread_data_t *data,
+					 sched_data_t *sched_data)
+{
+	log_debug("[%d] setting scheduler %s priority %d", data->ind,
+		  policy_to_string(__sched_policy(data, sched_data)),
+		  sched_data->prio);
+
+	log_ftrace(ft_data.marker_fd, FTRACE_ATTRS,
+		   "rtapp_attrs: event=policy policy=%s prio=%d",
+		  policy_to_string(__sched_policy(data, sched_data)),
+		   sched_data->prio);
+}
+
+static bool __set_thread_policy_priority(thread_data_t *data,
+					 sched_data_t *sched_data)
 {
 	struct sched_param param;
-	policy_t policy;
+	int ret;
+
+	param.sched_priority = __sched_priority(data, sched_data);
+
+	ret = pthread_setschedparam(pthread_self(),
+				    __sched_policy(data, sched_data),
+				    &param);
+	if (ret) {
+		log_critical("[%d] pthread_setschedparam returned %d",
+			     data->ind, ret);
+		errno = ret;
+		perror("pthread_setschedparam");
+		exit(EXIT_FAILURE);
+	}
+}
+
+static void __set_thread_nice(thread_data_t *data, sched_data_t *sched_data)
+{
+	int ret;
+
+	if (sched_data->prio > 19 || sched_data->prio < -20) {
+		log_critical("[%d] setpriority %d nice invalid. "
+			     "Valid between -20 and 19",
+			     data->ind, sched_data->prio);
+		exit(EXIT_FAILURE);
+	}
+
+	ret = setpriority(PRIO_PROCESS, 0, sched_data->prio);
+	if (ret != 0) {
+		log_critical("[%d] setpriority returned %d", data->ind, ret);
+		errno = ret;
+		perror("setpriority");
+		exit(EXIT_FAILURE);
+	}
+}
+
+static void _set_thread_cfs(thread_data_t *data, sched_data_t *sched_data)
+{
+	/* Priority unchanged => Policy unchanged */
+	if (sched_data->prio == THREAD_PRIORITY_UNCHANGED)
+		return;
+
+	/*
+	 * In the CFS case, sched_data->prio is the NICE value. As long as the
+	 * policy hasn't changed, there's no need to call
+	 * __set_thread_policy_priority()
+	 */
+	if (sched_data->policy == same)
```

Comment:
if (sched_data->policy != same) ?

Reply:
Woops :( Will fix that in the next version